tools/configen/configen/configen.py

```diff
@@ -129,16 +128,18 @@ def is_incompatible(type_: Type[Any]) -> bool:
     return True
 
 
-def get_default_flags(module: ModuleConf, imports_set: set) -> List[Parameter]:
+def get_default_flags(module: ModuleConf) -> List[Parameter]:
 
     def_flags: List[Parameter] = []
-    if module.default_flags:
+    if module.default_flags and hasattr(module.default_flags, "items"):
         for name, default in module.default_flags.items():
             if default:
                 flag_type = Flags.__annotations__[name].__args__[0].__name__
+                if "ConvertMode" in flag_type:
+                    flag_type = "str"
+                    default = str(default).strip("ConvertMode.").lower()
 
-                if name == "_convert_":
-                    imports_set.add(ConvertMode)
+                default = f'"{default}"' if flag_type == "str" else default
```

Comment:
no need for all of it.
There are exactly two flags in module.default_flags.{_recursive_, _convert_}.
just access them directly. no need for a generic loop and no need to deduce the type automatically.

Also, you can use default.name (and I don't think you want to lowercase).

Reply:
I thought the point was to write this generically so it can be added to in the future?

Also, it does need to be lower. It errors out if you pass 'ALL' for example.

Reply:
That was the initial idea, but once we have a dataclass describing the flags it's cleaner to just be explicit.

What error?

Reply:
```
 hydra.errors.InstantiationException: Error instantiating/calling 'torchvision.transforms.transforms.Compose' : Unsupported _convert_ value: ALL
``

Reply:
Is that the full error message?
This is really weird. ConvertMode is defined as:
```
class ConvertMode(Enum):
    NONE = 0
    PARTIAL = 1
    ALL = 2
```

All upper case.
I am expecting to b e able to use the string ALL as a valid value without having to lowercase it (which should result in an error).

Reply:
Okay, this is from _pop_convert_mode().
Can you also make it case insensitive? this is some custom logic. (normally OmegaConf enums are case sensitive).
In this case the implementation actually use lowercase even though the Enum is upper case, which is an error.

Reply:
Yeah. Thanks for tracking this down. I wasn't sure whether it was write to make up for this in configen or dive into hydras internals. Change is done.